design/gateway-apis-implementation.md

```diff
@@ -62,69 +61,77 @@ Project Contour (the organisation) may investigate a Gateway API based layer 4 s
 
 ### Contour Gateway model
 
-Contour considers a Gateway to describe a single Envoy deployment, deployed and managed by something outside of itself.
-Contour expects to be supplied with the full name (name and namespace) of a single Gateway, which it will watch and update the status of.
-This will be a configuration file parameter.
-When a Gateway is not supplied, Contour will not do Gateway API processing, or in other words, the existing `--experimental-service-apis` flag will be obsoleted and removed).
-Inside Contour will merge Listeners within its Gateway. (See the detailed design for the exact rules Contour will use for this.)
+Contour considers a Gateway to describe a single Envoy deployment.
+Contour expects to be supplied with a `ControllerName` which is required to match a `GatewayClass`.
+Contour will update the status of both the Gateway & GatewayClass.
+When a `ControllerName` is not supplied, Contour will not do Gateway API processing.
+Contour will merge Listeners within its Gateway. (See the detailed design for the exact rules Contour will use for this.)
+
+### Dynamic Envoy Configuration
+
+Typically, the fleet of Envoy's that connect to Contour to get their configuration information has been managed manually by the cluster operator.
+Given a more dynamic configuration API, it's convenient to have the Envoy fleet configuration in Kubernetes be managed by a controller.
+This allows cluster operator to simply deploy a Contour deployment, leaving the details about how Envoy is provisioned in the cluster as well as exposed outside of the cluster to a configuration file or CRD.
+
+This behavior is meant to be optional and layered, such that users who want a fully managed environment can deploy the Contour Operator which manages instances of Contour which in turn manages instances of Envoy.
+However, other users may not desire that level of controller driven infrastructure.
+This design leaves users with their choice of deployment models. 
```

Comment:
I think it'd be helpful to have a detailed design section that goes into this more since this is a big new feature. How does a user opt into getting the Envoy fleet automatically managed? How do they opt out? What are the configuration knobs/where are they specified? etc.